### PR TITLE
[Pods] Adding PodViewFilter

### PR DESCRIPTION
**Pull request by wavesoft**
This introduces a re-usable filter bar component that contains plain text search and buttons.

Example usage:

```jsx
  // Required UI State properties
  this.setState({
    filter: {
      text: '',
      status: ''
    }
  });

  ...

  // Status mapper is a function that receives an
  // item from the `items` array and returns
  // a string that represents the status of this item
  var statusMapper = function(item) {
    if (!item.active) {
      return 'inactive';
    }
    return 'active';
  }

  ...

  <PodViewFilter
    filter={this.state.filter}
    inverseStyle={inverseStyle}
    items={items}
    onFilterChange={this.handleFilterChange}
    statusChoices={['all', 'active', 'inactive']}
    statusMapper={statusMapper}
    />
```

### CODE REVIEW

File: src/js/components/PodViewFilter.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  getFilterButtonContent(filterName, count) {
+    return (
+      <span className="button-align-content">
+        <span className="label">{StringUtil.capitalize(filterName)}</span>
+        <span className="badge">{count || 0}</span>
+      </span>
+    );
+  }
+
+  getChildrenArray() {
```
Comment (mlunoe) [inline src/js/components/PodViewFilter.js]: This shouldn't be necessary, I think you can just add `{this.props.children}` in the render method

File: src/js/components/PodViewFilter.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  getFilterButtons() {
+    let {filter, items, inverseStyle, statusMapper, statusChoices} = this.props;
+    if (!statusChoices.length) {
+      return null;
+    }
+
+    return (
+        <FilterButtons
```
Comment (mlunoe) [inline src/js/components/PodViewFilter.js]: Indentation

File: src/js/components/PodViewFilter.js (hunk 1 above)
Comment (mlunoe) [inline src/js/components/PodViewFilter.js]: Because of this: https://github.com/dcos/dcos-ui/blob/master/src/js/components/FilterBar.js#L47

Comment (mlunoe) [inline src/js/components/PodViewFilter.js]: I removed this here: https://github.com/dcos/dcos-ui/pull/1069/commits/6254e5cad83be310776401bfb8beeff606b512a2

**Comment by mlunoe** [comment]
Thinking a bit about this afterwards I think we may be able to use this in multiple places, not only for pods. Could we create a ticket to roll this out for the rest of the application once `feature/pods` branch merges?

**Comment by wavesoft** [comment]
I thought so to, but I had some naming issues. For example the `FilterBar` is already taken. Perhaps we need to spend some additional thinking on this, but sure 👍

**Pull request merged**